job-dsls/jobs/compile_downstream_build.groovy

```diff
@@ -0,0 +1,206 @@
+/**
+ * Creates downstream pullrequest (PR) jobs for kiegroup GitHub org. units.
+ * These jobs execute the downstream build skipping tests and skipping the gwt compilation
+ * for a specific PR to make sure the changes do not break the downstream repos.
+ */
+import org.kie.jenkins.jobdsl.Constants
+
+def final DEFAULTS = [
+        ghOrgUnit              : Constants.GITHUB_ORG_UNIT,
+        branch                 : Constants.BRANCH,
+        timeoutMins            : 600,
+        label                  : "kie-rhel7 && kie-mem8g",
+        ghAuthTokenId          : Constants.GITHUB_AUTH_TOKEN,
+        upstreamMvnArgs        : "-B -e -T1C -DskipTests -Dgwt.compiler.skip=true -Dgwt.skipCompilation=true -Denforcer.skip=true -Dcheckstyle.skip=true -Dfindbugs.skip=true -Drevapi.skip=true clean install",
+        downstreamMvnGoals     : "-B -e -nsu -fae clean install -Dfull=true -DskipTests -Dgwt.compiler.skip=true -Dgwt.skipCompilation=true",
+        artifactsToArchive     : [
+                "**/target/*.log",
+                "**/target/testStatusListener*",
+                "**/target/screenshots/**",
+                "**/target/kie-wb*wildfly*.war",
+                "**/target/kie-wb*eap*.war",
+                "**/target/kie-drools-wb*wildfly*.war",
+                "**/target/kie-drools-wb*eap*.war",
+                "**/target/kie-server-*ee6.war",
+                "**/target/kie-server-*ee7.war",
+                "**/target/kie-server-*webc.war",
+                "**/target/jbpm-server*dist*.zip"
+        ],
+        excludedArtifacts      : [
+                "**/target/checkstyle.log"
+        ]
+]
+// override default config for specific repos (if needed)
+def final REPO_CONFIGS = [
+        "lienzo-core"               : [],
+        "lienzo-tests"              : [],
+        "kie-soup"                  : [],
+        "appformer"                 : [],
+        "droolsjbpm-build-bootstrap": [],
+        "droolsjbpm-knowledge"      : [],
+        "drlx-parser"               : [],
+        "drools"                    : [],
+        "optaplanner"               : [],
+        "jbpm"                      : [],
+        "droolsjbpm-integration"    : [],
+        //"droolsjbpm-tools"          : [], // no other repo depends on droolsjbpm-tools
+        "kie-uberfire-extensions"   : [],
+        "kie-wb-playground"         : [],
+        "kie-wb-common"             : [],
+        "drools-wb"                 : [],
+        "optaplanner-wb"            : [],
+        "jbpm-designer"             : [],
+        "jbpm-wb"                   : []
+        //"kie-wb-distributions"      : [] // kie-wb-distributions is the last repo in the chain
```

Comment:
Just out of curiosity, why is `kie-wb-distributions` commented out?

Reply:
@winklerm good question - this was taken 1:1 from Petr - so we did not change it. There should be no other rep dependent of kie-wb-distributions? (like droolsjbpm-tools)

Reply:
@winklerm To add my understanding: I believe it's ok to have `kie-wb-distributions` and `droolsjbpm-tools` commented out. 
The purpose of this groovy script is to generate "compile downstream" jobs is to make sure that changes in repo X don't break any of the repos that depend on X. In this particualr case there are no repos that depend on kie-wb-distributions and on droolsjbpm-tools, so we don't need jobs for those.